tensorboard/plugins/graph/polymer3/tf_graph_dashboard/tf-graph-dashboard.ts

```diff
@@ -0,0 +1,574 @@
+            selected-node="{{_selectedNode}}"
+            stats="[[_stats]]"
+            trace-inputs="[[_traceInputs]]"
+          ></tf-graph-board>
+        </div>
+      </tf-dashboard-layout>
+    </template>
+    <style>
+      :host /deep/ {
+        font-family: 'Roboto', sans-serif;
+      }
+
+      .sidebar {
+        display: flex;
+        height: 100%;
+      }
+
+      .center {
+        position: relative;
+        height: 100%;
+      }
+
+      paper-dialog {
+        padding: 20px;
+      }
+    </style>
+  `;
+  /**
+   * @type {!Array<!RunItem>}
+   */
+  @property({
+    type: Array,
+  })
+  _datasets: any[] = [];
+  @property({
+    type: Boolean,
+  })
+  _datasetsFetched: boolean = false;
+  @property({
+    type: Number,
+  })
+  _selectedDataset: number = 0;
+  @property({type: Object, observer: '_renderHierarchyChanged'})
+  _renderHierarchy: any;
+  @property({
+    type: Object,
+  })
+  _requestManager: RequestManager = new RequestManager();
+  @property({
+    type: Object,
+  })
+  _canceller: Canceller = new Canceller();
+  @property({type: Boolean})
+  _debuggerDataEnabled: boolean;
+  @property({type: Boolean})
+  allStepsModeEnabled: boolean;
+  @property({type: Number})
+  specificHealthPillStep: number = 0;
+  @property({
+    type: Boolean,
+    observer: '_healthPillsToggledOnChanged',
+  })
+  healthPillsToggledOn: boolean = false;
+  @property({
+    type: String,
+    notify: true,
+  })
+  selectedNode: string;
+  @property({type: Boolean})
+  _isAttached: boolean;
+  // Whether this dashboard is initialized. This dashboard should only be initialized once.
+  @property({type: Boolean})
+  _initialized: boolean;
+  // Whether health pills are currently being loaded, in which case we may want to say show a
+  // spinner.
+  @property({type: Boolean})
+  _areHealthPillsLoading: boolean;
+  // An array of alerts (in chronological order) provided by debugging libraries on when bad
+  // values (NaN, +/- Inf) appear.
+  @property({
+    type: Array,
+    notify: true,
+  })
+  _debuggerNumericAlerts: unknown[] = [];
+  // Maps the names of nodes to an array of health pills (HealthPillDatums).
+  @property({
+    type: Object,
+  })
+  _nodeNamesToHealthPills: object = {};
+  @property({type: Number})
+  _healthPillStepIndex: number;
+  // A strictly increasing ID. Each request for health pills has a unique ID. This helps us
+  // identify stale requests.
+  @property({type: Number})
+  _healthPillRequestId: number = 1;
+  /**
+   * The setTimeout ID for the pending request for health pills at a
+   * specific step.
+   *
+   * @type {number?}
+   */
+  @property({type: Number})
+  _healthPillStepRequestTimerId: number;
+  // The request for health pills at a specific step (as opposed to all sampled health pills) may
+  // involve slow disk reads. Hence, we throttle to 1 of those requests every this many ms.
+  @property({
+    type: Number,
+    readOnly: true,
```

Comment:
remove readOnly since the value here `500` won't really assign. Alternative is to provide `value: ` property in this `@property`, I think. Do a quick audit of all `readOnly` in the folder, too.